DependencyInjection/CmfRoutingExtension.php

```diff
@@ -319,4 +288,70 @@ public function getNamespace()
     {
         return 'http://cmf.symfony.com/schema/dic/routing';
     }
+
+    /**
+     * @param array $config
+     * @param ContainerBuilder $container
+     */
+    private function loadConditionalEnhancer(array $config, ContainerBuilder $container)
+    {
+        $enhancerMap = array();
+        $configurationPriorities = [
+            'controllers_by_type' => ['class' => FieldMapEnhancer::class, 'arguments' => ['type', '_controller']],
+            'controllers_by_class' => ['class' => FieldByClassEnhancer::class, 'arguments' => ['_content', '_controller']],
+            'templates_by_class' => ['class' => FieldByClassEnhancer::class, 'arguments' => ['_content', '_template']],
+        ];
+
+        foreach ($configurationPriorities as $key => $serviceDefinitionValues) {
+            if (count($config[$key]) === 0) {
+                continue;
+            }
+
+            foreach ($config[$key] as $mapKey => $mapping) {
+                $methods = is_array($mapping) && array_key_exists('methods', $mapping) ? $mapping['methods'] : null;
+                $value = is_array($mapping) && array_key_exists('value', $mapping) ? $mapping['value'] : $mapping;
+
+                $enhancerMap[] = [
+                    'matcher' => new RequestMatcher(null, null, $methods),
+                    'enhancer' => new $serviceDefinitionValues['class'](
+                        $serviceDefinitionValues['arguments'][0],
+                        $serviceDefinitionValues['arguments'][1],
+                        [$mapKey => trim($value)]
+                    ),
+                ];
+            }
+        }
+
+        if (0 < count($enhancerMap)) {
+            $container->getDefinition('cmf_routing.enhancer.conditional')->addArgument([]);#$enhancerMap);
```

Comment:
HI @WouterJ 
can you give me some help here, or just an advise. @dbu and i cam to the solution of an condional enhancer (symfony-cmf/routing#174) to handle method aware configuration. This one should get an enhancer map as created some lines above. This map unfortunatly contains objects, so i can't inject them as an argurment of a DI service defintion. You have an idea how to pass them in? One idea is to use an array only and build the object based map inside the enhancers constructor, which means to change that one in routing component, you have an other idea?